lang/python/python-paho-mqtt/Makefile

```diff
@@ -0,0 +1,58 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=python-paho-mqtt
+PKG_VERSION:=1.3.1
+PKG_RELEASE:=1
+PKG_MAINTAINER:=nobody
+PKG_LICENSE:=Eclipse Public License v1.0 / Eclipse Distribution License v1.0
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/eclipse/paho.mqtt.python
+PKG_SOURCE_VERSION:=v1.3.1
+
+PKG_SOURCE_SUBDIR:=$(BUILD_VARIANT)-paho-mqtt-$(PKG_VERSION)
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(BUILD_VARIANT)-paho-mqtt-$(PKG_VERSION)
+
+include $(INCLUDE_DIR)/package.mk
+include ../python-package.mk
+include ../python3-package.mk
+
+define Package/python-paho-mqtt/Default
+  SUBMENU:=Python
+  SECTION:=lang
+  CATEGORY:=Languages
+  MAINTAINER:=PierreF (https://github.com/PierreF)
+  URL:=http://eclipse.org/paho
+endef
+
+define Package/python-paho-mqtt
+$(call Package/python-paho-mqtt/Default)
+  TITLE:=python3-paho-mqtt
+  DEPENDS:=+python-light +python-idna +python-logging
```

Comment:
Please use ``DEPENDS:=+PACKAGE_python-paho-mqtt:python-light +PACKAGE_python-paho-mqtt:python-idna +PACKAGE_python-paho-mqtt:python-logging``  -- this prevents building both python and python3 if only one or the other is selected.

Reply:
hmm.  I had just copied what another python package was doing.  Can you send me a link to an example or instructions?  As-is, there is a separate package definition for `python-paho-mqtt` and `python3-paho-mqtt`.  Do you mean to move the DEPENDS to the `Package/python-paho-mqtt/Default` definition?  Thanks

Reply:
Actually I go better than that: @commodo added a doc describing the Python{3} build system here: https://github.com/openwrt/packages/blob/master/lang/python/README.md.  Also an examples that use this are: https://github.com/openwrt/packages/blob/master/lang/python/passlib/Makefile and https://github.com/openwrt/packages/blob/master/lang/python/python-dateutil/Makefile

Reply:
ok, I'm going to test this and see if it works:

```
define Package/python-paho-mqtt/Default
  SUBMENU:=Python
  SECTION:=lang
  CATEGORY:=Languages
  TITLE:=Eclipse Paho MQTT Python Client $(VARIANT)
  MAINTAINER:=Pierre Fersing <pierre.fersing@bleemeo.com>
  URL:=https://www.eclipse.org/paho/clients/python
  DEPENDS:= \
        +PACKAGE_$(VARIANT)-paho-mqtt:python-light \
        +PACKAGE_$(VARIANT)-paho-mqtt:python-idna \
        +PACKAGE_$(VARIANT)-paho-mqtt:python-logging
endef

define Package/python-paho-mqtt
$(call Package/python-paho-mqtt/Default)
  VARIANT:=python
endef

define Package/python3-paho-mqtt
$(call Package/python-paho-mqtt/Default)
  VARIANT:=python3
endef
```

Reply:
Hmmmm....you might need to move VARIANT:= above the $(call ... )

Reply:
oh!  also you'll need $(VARIANT)-light and so on...

Reply:
It'll be good to know if it works ...

Reply:
construct
```
  DEPENDS:= \
        +PACKAGE_$(VARIANT)-paho-mqtt:python-light \
        +PACKAGE_$(VARIANT)-paho-mqtt:python-idna \
        +PACKAGE_$(VARIANT)-paho-mqtt:python-logging
```
looks unsafe

because the depends is more like
```
  DEPENDS:= \
        +PACKAGE_python-paho-mqtt:python-light \
        +PACKAGE_python-paho-mqtt:python-idna \
        +PACKAGE_python-paho-mqtt:python-logging
```
and for Python3
```
  DEPENDS:= \
        +PACKAGE_python3-paho-mqtt:python3-light \
        +PACKAGE_python3-paho-mqtt:python3-idna \
        +PACKAGE_python3-paho-mqtt:python3-logging
```

while this can be updated, i still don't know how the order of evaluation of `PACKAGE_python-paho-mqtt` & `PACKAGE_python3-paho-mqtt` takes place when the build runs

`PACKAGE_python3-paho-mqtt` is a Kconfig symbol in the OpenWrt `.config` [which means this is `CONFIG_PACKAGE_python3-paho-mqtt` at some point in time] and there is a lot of black-magic happening behind the scenes with prefixing these variables;
so i would not parametrize this to avoid complications

Reply:
yep ; this form is pretty safe to do;
it's also a good example for others to copy if it can apply to their case [i.e. both python & python3 variants have the same depends)

Reply:
Makes sense to me; I was thinking of doing something about this duplication, but never got around to it.